test/c3/appframework/appframework_test.go

```diff
@@ -211,4 +212,83 @@ var _ = Describe("c3appfw test", func() {
+			appListV1 = testenv.ESApp
+			appFileList := testenv.GetAppFileList(appListV1, 1)
+
+			// Download V1 Apps from S3
+			err := testenv.DownloadFilesFromS3(testDataS3Bucket, s3AppDirV1, downloadDirV1, appFileList)
+			Expect(err).To(Succeed(), "Unable to download V1 app files")
+
+			// Upload V1 apps to S3
+			uploadedFiles, err := testenv.UploadFilesToS3(testS3Bucket, s3TestDir, appFileList, downloadDirV1)
+			Expect(err).To(Succeed(), "Unable to upload ES app to S3 test directory")
+			uploadedApps = append(uploadedApps, uploadedFiles...)
+
+			// Create App framework Spec
+			// volumeSpec: Volume name, Endpoint, Path and SecretRef
+			volumeName := "appframework-test-volume-" + testenv.RandomDNSName(3)
+			volumeSpec := []enterprisev1.VolumeSpec{testenv.GenerateIndexVolumeSpec(volumeName, testenv.GetS3Endpoint(), testenvInstance.GetIndexSecretName(), "aws", "s3")}
+
+			// AppSourceDefaultSpec: Remote Storage volume name and Scope of App deployment
+			appSourceDefaultSpec := enterprisev1.AppSourceDefaultSpec{
+				VolName: volumeName,
+				Scope:   "cluster",
+			}
+
+			// appSourceSpec: App source name, location and volume name and scope from appSourceDefaultSpec
+			appSourceName := "appframework-" + testenv.RandomDNSName(3)
+			appSourceSpec := []enterprisev1.AppSourceSpec{testenv.GenerateAppSourceSpec(appSourceName, s3TestDir, appSourceDefaultSpec)}
+
+			// appFrameworkSpec: AppSource settings, Poll Interval, volumes, appSources on volumes
+			appFrameworkSpec := enterprisev1.AppFrameworkSpec{
+				Defaults:             appSourceDefaultSpec,
+				AppsRepoPollInterval: 60,
+				VolList:              volumeSpec,
+				AppSources:           appSourceSpec,
+			}
+
+			// Create Single site Cluster and SHC, with App Framework enabled on CM and SHC Deployer
+			shSpec := enterprisev1.SearchHeadClusterSpec{
+				CommonSplunkSpec: enterprisev1.CommonSplunkSpec{
+					Spec: splcommon.Spec{
+						ImagePullPolicy: "Always",
+					},
+					ExtraEnv: []corev1.EnvVar{
+						{
+							Name:  "SPLUNK_ES_SSL_ENABLEMENT",
+							Value: "ignore"},
+					},
+					Volumes:                      []corev1.Volume{},
+					LivenessInitialDelaySeconds:  900,
+					ReadinessInitialDelaySeconds: 1500,
+				},
+				Replicas:           3,
+				AppFrameworkConfig: appFrameworkSpec,
+			}
+
+			_, err = deployment.DeploySearchHeadClusterWithGivenSpec(deployment.GetName()+"-shc", shSpec)
+			Expect(err).To(Succeed(), "Unable to deploy SHC with App framework")
+
+			// Ensure SHC go to Ready phase
+			testenv.SearchHeadClusterReady(deployment, testenvInstance)
+
+			// Verify ES app is installed locally on SHC Deployer
+			deployerPod := []string{fmt.Sprintf(testenv.DeployerPod, deployment.GetName())}
+			testenv.VerifyAppInstalled(deployment, testenvInstance, testenvInstance.GetName(), deployerPod, appListV1, false, "disabled", false, false)
+
+			// Verify ES app is installed on SHs
+			sh0 := fmt.Sprintf(testenv.SearchHeadPod, deployment.GetName(), 0)
```

Comment:
Maybe add a loop to get all the pod names? We set the replica factor to 3. so the same amount